Refactor: Simplify profile feature navigation

Joseph-Sameh-0 opened the pull request:
This commit refactors the navigation within the profile feature by:

- Removing the dedicated `ProfileNavGraph` and its associated navigator classes (`ProfileNavigator`, `ProfileNavigatorImpl`, `ProfileNavigationEvent`, etc.).
- Introducing a `ProfileDestinationActivity` to handle navigation to `WatchHistoryScreen` and `MyRatingScreen`.
- Navigation to these screens is now done by launching `ProfileDestinationActivity` with the target `Destination` serialized as a JSON string in the Intent extras.
- The `ProfileScreen` is now the direct entry point for the profile feature.
- `BaseViewModel` no longer holds a reference to `ProfileNavigator`.
- Updated `WatchHistoryScreen` and `MyRatingScreen` to use `LocalActivity.current.finish()` for back navigation instead of the ViewModel.
- Added serialization dependency for Kotlinx Serialization.
- Updated `ProfileViewModel` and `ProfileScreen` to use the new navigation mechanism.
- Removed `onBackClick` from `MyRatingViewModel`, `WatchHistoryViewModel`, and their respective interaction listeners.
- Moved `InstallSavedAppLanguage.kt` and `MainViewModel.kt` to a more general `main` package.
- Removed unused `BaseViewModel.kt` from `utils` package and `ObserveAsEvents.kt`.

Joseph-Sameh-0 committed:
Refactor: Simplify profile feature navigation

This commit refactors the navigation within the profile feature by:

- Removing the dedicated `ProfileNavGraph` and its associated navigator classes (`ProfileNavigator`, `ProfileNavigatorImpl`, `ProfileNavigationEvent`, etc.).
- Introducing a `ProfileDestinationActivity` to handle navigation to `WatchHistoryScreen` and `MyRatingScreen`.
- Navigation to these screens is now done by launching `ProfileDestinationActivity` with the target `Destination` serialized as a JSON string in the Intent extras.
- The `ProfileScreen` is now the direct entry point for the profile feature.
- `BaseViewModel` no longer holds a reference to `ProfileNavigator`.
- Updated `WatchHistoryScreen` and `MyRatingScreen` to use `LocalActivity.current.finish()` for back navigation instead of the ViewModel.
- Added serialization dependency for Kotlinx Serialization.
- Updated `ProfileViewModel` and `ProfileScreen` to use the new navigation mechanism.
- Removed `onBackClick` from `MyRatingViewModel`, `WatchHistoryViewModel`, and their respective interaction listeners.
- Moved `InstallSavedAppLanguage.kt` and `MainViewModel.kt` to a more general `main` package.
- Removed unused `BaseViewModel.kt` from `utils` package and `ObserveAsEvents.kt`.

mohammed-akkad commented on feature/profile/profileUi/src/main/java/com/feature/profile/profileUi/screen/ProfileViewModel.kt:
```diff
@@ -298,11 +300,11 @@ class ProfileViewModel @Inject constructor(
 
-    override fun onWatchHistoryClicked() {
-        navigate(ProfileDestinations.WatchHistoryScreen)
+    override fun onWatchHistoryClicked(context: Context) {
```
Why use context in view model?

Joseph-Sameh-0 replied:
I need it to navigate by intent

Joseph-Sameh-0 replied:
I just pass it

Joseph-Sameh-0 committed:
Refactor: Update Inject annotation from jakarta to javax

The `jakarta.inject.Inject` annotation has been replaced with `javax.inject.Inject` across multiple modules. This change affects ViewModels, DataSources, and Activities in the following features and data sources:
- lists
- search
- onboarding
- home
- profile
- mediaDetails
- authentication
- bottomNavBar
- user (local datasource)

Joseph-Sameh-0 committed:
Set ProfileDestinationActivity to not be exported

The `android:exported` attribute for `ProfileDestinationActivity` in the AndroidManifest.xml has been explicitly set to `false`. This prevents the activity from being launched by components of other applications.

Joseph-Sameh-0 committed:
Merge branch 'develop' into refactor/profile-feature-navigation

Joseph-Sameh-0 committed:
Bump version to 1.0.4
